benchmarks/README.md

```diff
@@ -0,0 +1,74 @@
+# README
+
+## Overview
+We use `asv` to benchmark some representative brainglobe workflows. The `asv` workflow is roughly as follows:
+1. `asv` creates a virtual environment (as defined in the `asv.conf.json` file).
+1. It installs the version of the `brainglobe-workflows` software package corresponding to the tip of the locally checked-out branch.
+1. It runs the benchmarks defined (locally) under `brainglobe-workflows/benchmarks/benchmarks` and saves the results to `brainglobe-workflows/benchmarks/results` as json files.
+1. With `asv publish`, the output json files are 'published' into an html directory (`brainglobe-workflows/benchmarks/html`)
+1. With `asv preview` the html directory can be visualised in a static site.
+
+
+There are three main ways in which these benchmarks can be useful to developers:
+1. Developers can run the available benchmarks locally [on a small test dataset](#running-benchmarks-locally-on-default-small-dataset).
+1. Developers can run these benchmarks on [data they have stored locally](#running-benchmarks-locally-on-custom-data).
+1. We also plan to run the benchmarks internally on a large dataset, and make the results publicly available.
+
+## Installation
+To [install asv](https://asv.readthedocs.io/en/stable/installing.html):
+```
+pip install asv
+```
+Note that to run the benchmarks you only need to have `asv` in your virtual environment. You _do not_ need to install a development version of `brainglobe-workflows`, since `asv` will create a separate Python virtual environment to run the benchmarks on it. However, for convenience we include `asv` as part of the `[asv]` dependencies, so an environment with `brainglobe-workflows[dev]` can be used to run the benchmarks.
+
+
+## Running benchmarks on a default small dataset
+
+
+1. Git clone the `brainglobe-workflows` repository
+    ```
+    git clone https://github.com/brainglobe/brainglobe-workflows.git
+    ```
+2. Install the `asv` version of the `brainglobe-workflows` package:
+    ```
+    pip install .[asv]
```

Comment:
Is it worth suggesting an editable install by default? To me it makes the most sense for a potential contributor.

Reply:
I think it may be confusing, because `asv` will create a separate environment and install the package from the selected commit/hash (by default, the checked out one). Installing or not the editable version of the package in the environment that launches `asv` won't change anything.

To make this more clear I modified the installation instructions to just `pip install asv` and removed the `[asv]` extra dependency option.

Reply:
> Installing or not the editable version of the package in the environment that launches asv won't change anything.

I just meant it's useful to add `-e` to have a consistent developer experience. We suggest they install packages this way [in the docs](https://brainglobe.info/community/developers/index.html#creating-a-development-environment), so they could wonder if this package is is somehow different. Assuming they're running benchmarks as part of the contributing process.

If they don't need to install, then this is moot :)